### PR TITLE
HBASE-26094 L2 BC should not be the victimhandler of L1 BC when using combined BC

### PR DESCRIPTION
**Pull request by YutSean**
https://issues.apache.org/jira/browse/HBASE-26094

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  16m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 45s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 55s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 54s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 14s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   4m  6s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 55s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 57s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   4m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 59s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 58s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 58s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m  4s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 54s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 56s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 53s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   3m 51s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 190m  1s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 258m  6s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.mapreduce.TestLoadIncrementalHFiles |
|   | hadoop.hbase.regionserver.TestSplitTransactionOnCluster |
|   | hadoop.hbase.mapreduce.TestLoadIncrementalHFilesUseSecurityEndPoint |
|   | hadoop.hbase.mapreduce.TestSecureLoadIncrementalHFiles |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3498 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 8e74a9e8cf07 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-3498/out/precommit/personality/provided.sh |
| git revision | branch-1 / edd6ea9 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/1/testReport/ |
| Max. process+thread count | 4538 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by YutSean** [comment]
The change passed TUs locally.

**Comment by Reidddddd** [comment]
Looks like 
```
// TODO: is there a hole here, or just awkwardness since in the lruCache getBlock
// we end up calling l2Cache.getBlock.
```
should these be removed in CombinedBlockCache together with this fix?

**Comment by YutSean** [comment]
Have removed the original comment and left a comment to mark that L2 is not the victim cache of L1 in CombinedBC.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java [hunk 1]
```diff
@@ -77,8 +77,7 @@ public void cacheBlock(BlockCacheKey cacheKey, Cacheable buf) {
   @Override
   public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching,
       boolean repeat, boolean updateCacheMetrics) {
-    // TODO: is there a hole here, or just awkwardness since in the lruCache getBlock
-    // we end up calling l2Cache.getBlock.
+    // After HBASE-26094, in CombinedBC L2 cache is not the victim handler of L1.
```
Comment (Reidddddd) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java:80]: unnecessary, better to remove it.

**Review by Reidddddd — APPROVED** [review]
+1, please fix the last comment

File: hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java (hunk 1 above)
Comment (YutSean) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java:80]: OK, will remove in the latest commit.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 50s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m  3s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 41s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 44s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  6s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  2s |  branch-1 passed  |
| -0 :warning: |  patch  |   3m 12s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 40s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 51s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 38s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 56s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 135m 46s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 184m 18s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestReplicationSource |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3498 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 82df984f9d1a 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3498/out/precommit/personality/provided.sh |
| git revision | branch-1 / b2d329b |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/3/testReport/ |
| Max. process+thread count | 4580 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 39s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 39s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 44s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 43s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 59s |  branch-1 passed  |
| -0 :warning: |  patch  |   3m  9s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 40s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 28s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 138m 10s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 39s |  The patch does not generate ASF License warnings.  |
|  |   | 179m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3498 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f3f9f32a4d6e 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3498/out/precommit/personality/provided.sh |
| git revision | branch-1 / b2d329b |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/4/testReport/ |
| Max. process+thread count | 4312 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3498/4/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by YutSean** [comment]
Any new review?

**Review by Reidddddd — APPROVED** [review]

**Pull request merged**